bin/docker-assets-precompile

```diff
@@ -16,6 +16,8 @@ rails_env = ENV.fetch('RAILS_ENV', 'production')
 if rails_env == 'production'
   env_keys = YAML.load_file('config/application.yml')['docker_build'].keys
   env_keys.each { |name| ENV[name] = 'dummy_value' }
```

Comment:
The current implementation works but can we modify this function like

```ruby
env_keys = YAML.load_file('config/application.yml')['docker_build'].keys
whitelist_envs = [DATABASE_URL]

env_keys.each do |name|
   if ![whitelist_envs].contain?(env) else
    ENV[name] = 'dummy_value'
   end
end
```

This allow us can extends if we would need to ignore any other new ENV, what do you think?

Reply:
cd3d640 ✅ 

Good idea! :)

Reply:
My purpose there was we don't need to re-set the `DATABASE_URL` manually and keep the original value for that.

But I just checked the `config/application.yml`['docker_build'] config; seems we don't have the default value for the `DATABASE_URL`, then the `whitelist` here doesn't bring any sense 😂

Another idea, do you think we can create a new scope in the `config/application.yml` that extends from the `docker_build` 🤔 to support precompiling the asset 🤔


```ruby
docker_precompile_asset:
  <<: *docker_build
  DATABASE_URL:  postgres://postgres:postgres@postgres:5432/postgres
```

Or we can set the DATABASE_URL directly in the https://github.com/nimblehq/rails-templates/blob/develop/config/application.yml.tt as it already mentions that is mostly used for the `bin/docker-assets-precompile`

![image](https://user-images.githubusercontent.com/11751745/187141763-105e0b01-77fe-4766-b0f1-dc624e08b426.png)

Reply:
IMO we need to work on a cleaner long-term fix for this workaround of `precompile asset script`. 
So I think we should not change any other file for the sake of cleaning this temporary script 🤔

Reply:
There is already [an issue with that](https://github.com/nimblehq/rails-templates/issues/326) and it is not the purpose of this PR (this PR aims to fix a bug in fact ^^)

Reply:
Then could you remove the `whitelist_envs` solution above @malparty and put a TODO link to the https://github.com/nimblehq/rails-templates/issues/326 as we will improve that there 🤔

Reply:
60189c6